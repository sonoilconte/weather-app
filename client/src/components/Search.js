import React from 'react';

class Search extends React.Component {
	render() {
		return (
			<div className="row justify-content-center">
				<div className='col-md-6'>
					<form onSubmit={this.props.onSearchTermSubmit}>
						<input
							placeholder='Enter a city or zip code'
							onChange={this.props.onSearchTermChange}
						/>
						<input type='submit' className="btn btn-primary" />
					</form>
				</div>
			</div>
		);
	}
}

export default Search;