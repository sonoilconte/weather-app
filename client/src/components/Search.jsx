import React from 'react';
import PropTypes from 'prop-types';

function Search({ onSearchTermChange, onSearchTermSubmit }) {
    return (
        <div className="row justify-content-center">
            <div className="col-md-6">
                <form onSubmit={onSearchTermSubmit}>
                    <input
                        placeholder="Enter a city or zip code"
                        onChange={onSearchTermChange}
                        data-testid="searchField"
                    />
                    <input
                        type="submit"
                        className="btn btn-primary"
                        data-testid="submitBtn"
                    />
                </form>
            </div>
        </div>
    );
}

Search.propTypes = {
    onSearchTermChange: PropTypes.func.isRequired,
    onSearchTermSubmit: PropTypes.func.isRequired,
};

export default Search;
