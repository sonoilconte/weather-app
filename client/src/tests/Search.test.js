import renderer from 'react-test-renderer';

import Search from '../components/Search';

it('renders the Search component correctly', () => {
	const tree = renderer.create(<Search />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});