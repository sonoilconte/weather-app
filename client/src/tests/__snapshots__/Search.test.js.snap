// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Search component correctly 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-md-6"
  >
    <form>
      <input
        placeholder="Enter a city or zip code"
      />
      <input
        className="btn btn-primary"
        type="submit"
      />
    </form>
  </div>
</div>
`;
